File: Settings/src/org/nameless/settings/display/PerAppAutoRotateFragment.java
/*
 * Copyright (C) 2023-2024 The Nameless-AOSP Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.nameless.settings.display;

import static org.nameless.os.RotateManager.ROTATE_FOLLOW_SYSTEM;
import static org.nameless.os.RotateManager.ROTATE_FORCE_OFF;
import static org.nameless.os.RotateManager.ROTATE_FORCE_ON;

import android.os.Bundle;

import com.android.settings.R;

import java.util.ArrayList;
import java.util.List;

import org.nameless.os.RotateManager;
import org.nameless.settings.fragment.PerAppListConfigFragment;

public class PerAppAutoRotateFragment extends PerAppListConfigFragment {

    private RotateManager mRotateManager;

    private List<String> mEntries;
    private List<Integer> mValues;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        mRotateManager = getActivity().getSystemService(RotateManager.class);

        mEntries = new ArrayList<>();
        mEntries.add(getString(R.string.per_app_auto_rotate_default));
        mEntries.add(getString(R.string.per_app_auto_rotate_off));
        mEntries.add(getString(R.string.per_app_auto_rotate_on));

        mValues = new ArrayList<>();
        mValues.add(ROTATE_FOLLOW_SYSTEM);
        mValues.add(ROTATE_FORCE_OFF);
        mValues.add(ROTATE_FORCE_ON);
    }

    @Override
    protected int getPreferenceScreenResId() {
        return R.xml.per_app_auto_rotate_config;
    }

    @Override
    protected int getTopInfoResId() {
        return R.string.per_app_auto_rotate_summary;
    }

    @Override
    protected List<String> getEntries() {
        return mEntries;
    }

    @Override
    protected List<Integer> getValues() {
        return mValues;
    }

    @Override
    protected int getCurrentValue(String packageName, int uid) {
        return mRotateManager.getRotateConfigForPackage(packageName);
    }

    @Override
    protected void onValueChanged(String packageName, int uid, int value) {
        mRotateManager.setRotateConfigForPackage(packageName, value);
    }
}

File: Settings/src/org/nameless/settings/display/PerAppAutoRotateFragment.kt
/*
 * Copyright (C) 2025 The Nameless-CLO Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.nameless.settings.display

import com.android.settings.R

import org.nameless.os.RotateManager
import org.nameless.os.RotateManager.ROTATE_FOLLOW_SYSTEM
import org.nameless.os.RotateManager.ROTATE_FORCE_OFF
import org.nameless.os.RotateManager.ROTATE_FORCE_ON
import org.nameless.settings.fragment.PerAppListConfigFragment

class PerAppAutoRotateFragment : PerAppListConfigFragment() {

    private val rotateManager by lazy { requireContext().getSystemService(RotateManager::class.java)!! }

    override fun getTitleResId() = R.string.per_app_auto_rotate_title

    override fun getTopInfoResId() = R.string.per_app_auto_rotate_summary

    override fun getEntries() = listOf(
        getString(R.string.per_app_auto_rotate_default),
        getString(R.string.per_app_auto_rotate_off),
        getString(R.string.per_app_auto_rotate_on)
    )

    override fun getValues() = listOf(
        ROTATE_FOLLOW_SYSTEM,
        ROTATE_FORCE_OFF,
        ROTATE_FORCE_ON
    )

    override fun getCurrentValue(packageName: String, uid: Int): Int {
        return rotateManager.getRotateConfigForPackage(packageName)
    }

    override fun getDefaultValue() = ROTATE_FOLLOW_SYSTEM

    override fun onValueChanged(packageName: String, uid: Int, value: Int) {
        rotateManager.setRotateConfigForPackage(packageName, value)
    }
}
